drivers/serial/uart_sam0.c

```diff
@@ -71,6 +72,8 @@ struct uart_sam0_dev_data {
 #endif
 };
 
+static struct uart_config config_cache;
```

Comment:
This is incorrect -- it cannot support multiple UART device instances.

The `config_cache` should be part of the device context (i.e. `struct uart_sam0_dev_cfg` or `struct uart_sam0_dev_data`) to store the config values for each UART device instance.

Reply:
Done.